# HG changeset patch
# User Carlo de Falco <carlo.defalco@polimi.it>
# Date 1450291199 -3600
#      Wed Dec 16 19:39:59 2015 +0100
# Node ID c19f128ea6df39ec32941613e7ddf137a5ca3408
# Parent  359476667c4ccac13a46d08e3f57f0595b7e76e2
add file missing from commit

* scripts/ode/private/known_option_names.m: new file.

diff --git a/scripts/ode/private/known_option_names.m b/scripts/ode/private/known_option_names.m
new file mode 100644
--- /dev/null
+++ b/scripts/ode/private/known_option_names.m
@@ -0,0 +1,35 @@
+## Copyright (C) 2015, Carlo de Falco
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn  {} {@var{val} =} known_option_names ()
+## Return a list of known names for ode options.
+## @seealso{odeset,odeget}
+## @end deftypefn
+
+function ret = known_option_names ()
+
+ret = {"AbsTol"; "BDF"; "Events"; "InitialSlope";
+       "InitialStep"; "Jacobian"; "JConstant"; "JPattern";
+       "Mass"; "MassConstant"; "MassSingular"; "MaxOrder";
+       "MaxStep"; "MStateDependence"; "MvPattern";
+       "NonNegative"; "NormControl"; "OutputFcn"; "OutputSel";
+       "Refine"; "RelTol"; "Stats"; "Vectorized";
+       "TimeStepSize"; "TimeStepNumber"};
+
+endfunction
